cluster-autoscaler/cloudprovider/gce/autoscaling_gce_client.go

```diff
@@ -425,6 +425,7 @@ func createIgmRef(gceInstance *gce.Instance, loggingQuota *klogx.Quota) GceRef {
 		klogx.V(5).UpTo(loggingQuota).Infof("Unable to parse IGM for %v because of %v", gceInstance.SelfLink, err)
 		return GceRef{}
 	}
+	igmRef.Project = project
```

Comment:
Please leave a comment on why we need to overwrite the project. I'm wondering also if it would be better for [ParseIgmUrl](https://github.com/kubernetes/autoscaler/blob/35359b902ba00931fcad5be7c72a03bd7be39f7f/cluster-autoscaler/cloudprovider/gce/gce_url.go#L44) to only fill in zone and igm name and not the project. WDYT?

Reply:
Added a comment to describe why we overwrite the project.
 
What do you think of keeping the PraseIgmUrl as it is since it is a generic function that somebody can use elsewhere?